### ISSUE TITLE
[Localization] Translation of components doesn't work with Angular 17

### ISSUE THREAD

**Issue by yanmariomenev** [telerik/kendo-angular#4195]
**Describe the bug**
Translation with Kendo UI for Angular components fails when using Angular 17. It seems that the issue is related to changes in the localize version as it works with Angular v16.

**To Reproduce**
Run the following app:

[Archive.zip](https://github.com/telerik/kendo-angular/files/13938857/Archive.zip)

ng serve --configuration=es

**Comment by JeanMeche** [comment]
I just took a look at it. 
The v16 version updated via `ng update` to v17 run fine. 

The v17 project itself is a bit different as it uses the application builder (esbuild) instead of the webpack builder.

**Comment by alyssamichelle** [comment]
Same, I tried updating both the v17 version and the v16 to the latest (v17.1.2) and I think they are both working as intended.. unless I am missing something. 

<img width="1057" alt="image" src="https://github.com/telerik/kendo-angular/assets/1058831/ddb5fa1b-959c-4999-900c-9825ba4de4c1">

[a16-updated.zip](https://github.com/telerik/kendo-angular/files/14145008/a16-updated.zip)

![image](https://github.com/telerik/kendo-angular/assets/1058831/0254b5e6-cf9f-405a-ba95-6138f97d6ad4)

**Comment by Julian-B90** [comment]
@alyssamichelle yes your are right but you have so "old" builder with Webpack noch Vite+esbuild. With esbuild and vite it doesn´t work.

**Comment by kanishkAsharm** [comment]
> **Describe the bug** Translation with Kendo UI for Angular components fails when using Angular 17. It seems that the issue is related to changes in the localize version as it works with Angular v16.
> 
> **To Reproduce** Run the following app:
> 
> [Archive.zip](https://github.com/telerik/kendo-angular/files/13938857/Archive.zip)
> 
> Working Angular 16 project:
> 
> [a16.zip](https://github.com/telerik/kendo-angular/files/13947277/a16.zip)
> 
> run command - `ng serve --configuration=es`



> Same, I tried updating both the v17 version and the v16 to the latest (v17.1.2) and I think they are both working as intended.. unless I am missing something. @stnikolova
> 
> <img alt="image" width="1057" src="https://private-user-images.githubusercontent.com/1058831/301962835-ddb5fa1b-959c-4999-900c-9825ba4de4c1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxNzE4MDEsIm5iZiI6MTcxMzE3MTUwMSwicGF0aCI6Ii8xMDU4ODMxLzMwMTk2MjgzNS1kZGI1ZmExYi05NTljLTQ5OTktOTAwYy05ODI1YmE0ZGU0YzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMDg1ODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWEzNmE3YmFkOTdhYWRkNGJiOTFjZjcxMTExYjUyZjZhZDdkNjAwZjk4ZGQ4NjcwOTliMDFhMGRkYjVkN2JlOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RCn87OJ9sfCsiDSdpaGzzqrhEAR3P3u57S9WiRThyfQ">
> [a16-updated.zip](https://github.com/telerik/kendo-angular/files/14145008/a16-updated.zip)
> 
> ![image](https://private-user-images.githubusercontent.com/1058831/301961312-0254b5e6-cf9f-405a-ba95-6138f97d6ad4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxNzE4MDEsIm5iZiI6MTcxMzE3MTUwMSwicGF0aCI6Ii8xMDU4ODMxLzMwMTk2MTMxMi0wMjU0YjVlNi1jZjlmLTQwNWEtYmE5NS02MTM4Zjk3ZDZhZDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMDg1ODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2MwNTJiOTJlYjUyNDA0ZmVkYWFiZWZmNDFlMjVmMjFjNTkwY2U0ZmE3ZGVmY2MxYjc1MzBmYzA2NGEyNmZiNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.iT3kD2uIPcyGS_5oGRtwmYn8erIDcOj2C5TqgpVVfUQ)

How to run this application to switch the laguages??

**Comment by OrForte** [comment]
Hi @stnikolova,
Unfortunately, running the command `ng serve --configuration=es` doesn't change the language to Spanish.
Is there a workaround to make the localization work on Angular 17 with esbuild?
Thanks.
